### PR TITLE
the last big bang release

### PR DESCRIPTION
**Pull request by jurgenvinju**
* [ ] bumped dependencies on rascal, rascal-maven-plugin and bumped all project versions and released all packages:
   * [ ] rascal
   * [ ] rascal-lsp
   * [x] rascal-tutor
   * [x] flybytes
   * [ ] clair
   * [ ] salix
   * [ ] salix-contrib
   * [ ] rascal-git
   * [x] typepal
   * [x] drambiguity
   * [ ] php-analysis
   * [x] python-air
* [x] documentation still checks without errors
* [ ] documentation still renders without errors
   * [x] problems in rascal-tutor
   * [x] problems in rascal library
   * [x] problems in flybytes
   * [x] problems in typepal
* [ ] release notes in order
   * [ ] rascal
   * [ ] rascal-language-servers
   * [ ] rascal-lsp
* [ ] rascal-language-servers released
* [x] fix duplicate rascal library documentation

### CODE REVIEW

File: release-notes/rascal-0-40-x-release-notes.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+A great deal of tests were fixed, enhanced or extended as a side-effect of the compiler project. Another visible aspect of the progress of the compiler is that now all Rascal runtime values (from vallang and beyond) now support `getFingerprint()` methods which help in optimizing patter matching and dispatch in generated code by the compiler. These  methods' return values have become a strict contract for future implementations of Rascal values, including parse trees and reified types and first-class functions.
+
+The following issues were solved: 
+* string `visit` with unicode characters had a bug
+* `util::ShellExec` had some IO synchronization issues which were resolved.
+* The JUnit test runners report exceptions better now.
+* The broken and unused `stdout://` and `stdin://` resolvers were removed.
+* The `IO::watch` functionality was improved for logical source locations. However the system is still too slow on Mac to work effectively in an interactive development environment.
```
Comment (DavyLandman) [inline release-notes/rascal-0-40-x-release-notes.md]: maybe point out that the mac thing is a basic jdk limitation of the polling mechanism? and that it's still better than nothing.

File: release-notes/rascal-0-40-x-release-notes.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+* TODO
```
Comment (DavyLandman) [inline release-notes/rascal-0-40-x-release-notes.md]: maybe also include at least one line about various the performance fixes?

File: release-notes/rascal-0-40-x-release-notes.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+sidebar_position: 96
+---
+
+In this post we report on the Rascal release 0.40.x
+
+<!--truncate-->
+
+## Release 0.40.14 - November 5, 2024
+
+The public release 0.40.x follows release 0.28.x; many improvements have been made in projects that depends on the rascal interpreter and the standard library (the type checker, the VScode extensions, clair, etc.) Some of these improvements depend directly on fixes in the interpreter and additions to the standard library.
```
Comment (DavyLandman) [inline release-notes/rascal-0-40-x-release-notes.md]: We need to have a paragraph that describes the new tpl&checker situation. What users will have to do, and what in the future they don't have to anymore.

**Review by DavyLandman — COMMENTED** [review]
What a work to make that release note! I've added a few points.

File: release-notes/rascal-0-40-x-release-notes.md (hunk 1 above)
Comment (jurgenvinju) [reply release-notes/rascal-0-40-x-release-notes.md]: Super!

File: release-notes/rascal-0-40-x-release-notes.md (hunk 2 above)
Comment (jurgenvinju) [reply release-notes/rascal-0-40-x-release-notes.md]: Just in general? I don't have the specifics ready

File: release-notes/rascal-0-40-x-release-notes.md (hunk 3 above)
Comment (jurgenvinju) [reply release-notes/rascal-0-40-x-release-notes.md]: you're right! That's not this package but it is Rascal in general. I'll have a look also at the maven plugin diff.

**Pull request merged**